### PR TITLE
fix(webdriver): emit RequestServedFromCache for requests

### PR DESCRIPTION
**Pull request by Lightning00Blade**

### CODE REVIEW

File: packages/puppeteer-core/src/bidi/HTTPRequest.ts [hunk 1]
```diff
@@ -68,7 +68,7 @@ export class BidiHTTPRequest extends HTTPRequest {
   }
 
   override resourceType(): ResourceType {
-    return this.initiator().type.toLowerCase() as ResourceType;
```
Comment (Lightning00Blade) [inline packages/puppeteer-core/src/bidi/HTTPRequest.ts:71]: This is totally wrong... We should see if we can add `ResourceType` to the spec. 
Both Firefox and Chrome support similar functionality in there respective DevTools panels.

**Review by OrKoN — APPROVED** [review]
LGTM % remove the fail expectation

**Comment by Lightning00Blade** [comment]
@OrKoN Please rereview I need to fix the expectation formatter.

File: tools/sort-test-expectations.mjs [hunk 2]
```diff
@@ -92,14 +92,17 @@ for (let i = testExpectations.length - 1; i >= 0; i--) {
       testIdMatchesExpectationPattern(
         expectation.testIdPattern,
         candidate.testIdPattern
-      ) &&
-      isSubset(candidateParams, params) &&
-      isSubset(candidatePlatforms, platforms)
+      )
     ) {
       foundMatch = true;
```
Comment (Lightning00Blade) [inline tools/sort-test-expectations.mjs]: We need to mark that there is a match even if the other parts don't match.
```json
  {
    "testIdPattern": "[network.spec] network Network Events Page.Events.RequestServedFromCache",
    "platforms": ["darwin", "linux", "win32"],
    "parameters": ["chrome", "webDriverBiDi"],
    "expectations": ["PASS"]
  },
  ```
  Was being remove due to 
  ```json
    {
    "testIdPattern": "[network.spec] network Network Events *",
    "platforms": ["darwin", "linux", "win32"],
    "parameters": ["webDriverBiDi"],
    "expectations": ["FAIL"],
    "comment": "TODO: add a comment explaining why this expectation is required (include links to issues)"
  },
  ```

**Review by OrKoN — APPROVED** [review]

File: tools/sort-test-expectations.mjs (hunk 2 above)
Comment (OrKoN) [reply tools/sort-test-expectations.mjs]: Aren't params needed for the match too?

File: tools/sort-test-expectations.mjs [hunk 3]
```diff
@@ -93,13 +93,14 @@ for (let i = testExpectations.length - 1; i >= 0; i--) {
...
       ) &&
-      isSubset(candidateParams, params) &&
       isSubset(candidatePlatforms, platforms)
     ) {
       foundMatch = true;
-      if (isSubset(candidateLabels, labels)) {
-        console.log('removing', expectation, 'already covered by', candidate);
-        toBeRemoved.add(expectation);
+      if (isSubset(candidateParams, params)) {
+        if (isSubset(candidateLabels, labels)) {
```
Comment (OrKoN) [inline tools/sort-test-expectations.mjs]: merge ifs?

File: tools/sort-test-expectations.mjs (hunk 2 above)
Comment (Lightning00Blade) [reply tools/sort-test-expectations.mjs]: Yeah, I think the logic is wrong.
I think the `isSubset(candidateParams, params)` need to be reverse as we have an `AND` logic for Params

So previous expectation will have a smaller `Params` then the next one and still affect it.

**Comment by Lightning00Blade** [comment]
@OrKoN after the last change this was the output - https://gist.github.com/Lightning00Blade/50916ff9d8926e5e3fb629ad4311a18a

File: tools/sort-test-expectations.mjs (hunk 2 above)
Comment (OrKoN) [reply tools/sort-test-expectations.mjs]: I am not sure that is correct, reversing might remove a more specific expectations I think

File: log.log [hunk 4]
```diff
@@ -0,0 +1,169 @@
+
```
Comment (OrKoN) [inline log.log]: remove

**Review by OrKoN — APPROVED** [review]

**Pull request merged**